Sources/Workspace/Workspace.swift

```diff
@@ -293,183 +301,218 @@ public class Workspace {
     private func createCacheDirectories(diagnostics: DiagnosticsEngine) {
         do {
             try fileSystem.createDirectory(repositoryManager.path, recursive: true)
+        } catch {
+            diagnostics.emit(error)
+        }
+        
+        do {
             try fileSystem.createDirectory(checkoutsPath, recursive: true)
         } catch {
             diagnostics.emit(error)
         }
     }
 
     /// Cleans the build artefacts from workspace data.
-    public func clean() throws {
-        // These are the things we don't want to remove while cleaning.
-        let protectedAssets = Set<String>([
-            repositoryManager.path,
-            checkoutsPath,
-            try managedDependencies.load().statePath
-        ].map { path in
-            // Assert that these are present inside data directory.
-            assert(path.parentDirectory == dataPath)
-            return path.basename
-        })
-        // If we have no data yet, we're done.
-        guard fileSystem.exists(dataPath) else {
-            return
-        }
-        for name in try fileSystem.getDirectoryContents(dataPath) {
-            guard !protectedAssets.contains(name) else { continue }
-            fileSystem.removeFileTree(dataPath.appending(RelativePath(name)))
+    ///
+    /// - Parameters:
+    ///     - diagnostics: The diagnostics engine that reports errors, warnings
```

Comment:
I feel sad when the documentation for a parameter just repeats the words in the type. Isn't it worth keeping this slightly longer description in case this is the first place a user sees the diagnostics engine mentioned so he understands what it does?